src/FormBuilder/resources/js/form-builder/src/settings/donation-goal/index.tsx

```diff
@@ -64,7 +68,15 @@ const DonationGoalSettings = () => {
                             label={__('Goal Type', 'give')}
                             value={goalType}
                             options={goalTypeOptions}
-                            onChange={(goalType) => dispatch(setFormSettings({goalType: goalType}))}
+                            onChange={(goalType: string) => {
+                                dispatch(setFormSettings({goalType}))
+                                publish('preview:goal', {
+                                    type: goalType,
+                                    label: goalType,
+                                    typeIsCount: 'amount' !== goalType,
+                                    typeIsMoney: 'amount' === goalType
+                                }, iframeRef);
+                            }}
```

Comment:
I'm looking at this thinking there is a lot of contexts to understand about adding a form setting.  We have the form setting json and now window data object to remember.  I would like to explore abstracting the publishing of preview data.  At the very least, we should add some extra functions like `publishPreviewGoal(data: Form['goal'])` and take advantage of typescript for the accepted data.

I'm open to ideas. Part of me thinks this could be a job for the reducer to handle since it's the closest modifier to the form's state but maybe that's a separation of concerns issue.

Reply:
I was thinking of adding a helper function, and we should probably do that. I think using reducers will add unnecessary overhead, with little or no benefit.

Reply:
helper functions would probably suffice.  the reducer I was referring to was the one we are already using for setting the values - but again that might be better seperated.